src/main/java/com/atlassian/bitbucket/jenkins/internal/client/HttpRequestExecutor.java

```diff
@@ -45,6 +45,25 @@ public interface HttpRequestExecutor {
      */
     <T> T executeGet(HttpUrl url, ResponseConsumer<T> consumer, RequestConfiguration... additionalConfig);
 
+    /**
+     * Executes a streaming get call to a given URL.
+     *
+     * @param url              The URL to hit on bitbucket server end
+     * @param additionalConfig additional configuration, such as credentials, that will be used in making calls
+     * @param consumer         on successful execution, {@link Response} will be passed to consumer
+     * @param <T>              result that consumer wish to return
+     * @return result
+     * @throws AuthorizationException     if the credentials did not allow access to the given url
+     * @throws ConnectionFailureException if the server did not respond
+     * @throws NotFoundException          if the requested url does not exist
+     * @throws BadRequestException        if the request was malformed and thus rejected by the server
+     * @throws ServerErrorException       if the server failed to process the request
+     * @throws BitbucketClientException   for all errors not already captured
+     *
+     * @since 4.43
+     */
+    <T> T executeStreamingGet(HttpUrl url, ResponseConsumer<T> consumer, RequestConfiguration... additionalConfig);
```

Comment:
Hmm.. wouldn't this _always_ return an `InputStream`? How could I be streamlining and not return a stream? I think this signature needs to change to 
`InputStream executeStreamingGet(HttpUrl url,  RequestConfiguration... additionalConfig);`

The implementation throws away any knowledge of `T` and also throws away the `consumer`  without using it.